### PR TITLE
Data Explorer: Address DuckDB identifier quoting issues with fields containing non-ASCII characters 

### PR DESCRIPTION
**Pull request by wesm**
Addresses #7958. There were many places where a field identifier with non-ASCII characters or quotations (double or single quotes) would causes error's in DuckDB's SQL parser. This adds the appropriate quoting and adds more robust logic in the generation of auxiliary field names for the various statistics that are computed to power the summary pane. 

e2e: @:data-explorer

## QA Notes

I checked the various files provided in https://github.com/posit-dev/positron/issues/7958 and verified that they can be loaded well now

image

### CODE REVIEW

File: extensions/positron-duckdb/src/extension.ts [hunk 1]
```diff
@@ -248,16 +248,50 @@ function makeWhereExpr(rowFilter: RowFilter): string {
...
+ * Returns a valid SQL identifier even for empty strings.
+ * @param fieldName The field name to quote
+ * @returns The properly quoted and escaped identifier
+ */
 function quoteIdentifier(fieldName: string) {
-	return `"${fieldName}"`;
+	// Handle empty string case by returning a valid SQL identifier
+	if (fieldName === '') {
+		return '""';
+	}
```
Comment (austin3dickey) [inline extensions/positron-duckdb/src/extension.ts]: nit: is this `if` statement redundant?

**Review by austin3dickey — DISMISSED** [review]
Nice, it worked for me too! I'm surprised you didn't have to change any test cases, do we not have coverage of these functions?

**Comment by wesm** [comment]
I will look at adding some extension unit tests for field names with quotes, historically most of the testing for this feature has taken place at the e2e / QA level and the tests that are there serve mainly as a smoke test for some of the calculations (e.g. the histogram calculations) and the formatting of data values.

File: extensions/positron-duckdb/src/extension.ts (hunk 1 above)
Comment (wesm) [reply extensions/positron-duckdb/src/extension.ts]: I'll take a closer look and maybe add some tests for the scenario where a column name is an empty string

**Comment by wesm** [comment]
I added a bunch of AI-assisted unit tests, so the coverage is pretty solid now
